src/main/java/com/takipi/oss/storage/fs/cache/InMemoryCache.java

```diff
@@ -0,0 +1,68 @@
+package com.takipi.oss.storage.fs.cache;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class InMemoryCache implements Cache
+{
+	private static final Logger logger = LoggerFactory.getLogger(InMemoryCache.class);
+	
+	private final Map<String, String> cache;
+	
+	public InMemoryCache(int maxCacheSize) {
+		
+		int minAllowedCacheSize = 65536;
+		int maxAllowedCacheSize = 134217728;
+		
+		if (maxCacheSize < minAllowedCacheSize) {
+			logger.warn("Minimum allowable In Memory Cache size = ", minAllowedCacheSize);
+			maxCacheSize = minAllowedCacheSize;
+		}
+		else if (maxCacheSize > maxAllowedCacheSize) {
+			logger.warn("Maximum allowable In Memory Cache size = ", maxAllowedCacheSize);
+			maxCacheSize = maxAllowedCacheSize;
+		}
+		
+		final int cacheSizeLimit = maxCacheSize;
+		logger.info("In Memory Cache maximum size = " + maxCacheSize);
+		
+		int estimatedSizePerElement = 600;
+		int initialCapacity = (int)Math.pow(2, Math.ceil(Math.log((float)maxCacheSize / estimatedSizePerElement) / Math.log(2)));
+		
+		cache = Collections.synchronizedMap(
```

Comment:
why synchronized map?

Reply:
There is no concurrent LinkedHashMap that I know of (please let me know if there's something I don't know about).
Anyway, it's unlikely that there will be thread contention on this map that will have a noticeable impact on the performance. Probably the thread contention will be very low.